cmd/dmverity-vhd/main.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/Microsoft/hcsshim/ext4/tar2ext4"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+var (
+	image    = flag.String("i", "", "image")
+	output   = flag.String("o", "", "output")
+	verbose  = flag.Bool("v", false, "verbose")
+	username = flag.String("u", "", "username")
+	password = flag.String("p", "", "password")
+)
+
+func main() {
+	flag.Parse()
+	if flag.NArg() != 0 || len(*image) == 0 || len(*output) == 0 {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	err := func() (err error) {
+		ref, err := name.ParseReference(*image)
+		if err != nil {
+			fmt.Printf("'%s' isn't a valid image name\n", *image)
+			os.Exit(1)
+		}
+
+		var imageOptions []remote.Option
+		if len(*username) != 0 && len(*password) != 0 {
+			auth := authn.Basic{
+				Username: *username,
+				Password: *password}
+			c, _ := auth.Authorization()
+			authOption := remote.WithAuth(authn.FromConfig(*c))
+			imageOptions = append(imageOptions, authOption)
+		}
+
+		img, err := remote.Image(ref, imageOptions...)
+		if err != nil {
+			fmt.Printf("Unable to fetch image '%s'\n", *image)
+			fmt.Printf("%s\n", err.Error())
+			os.Exit(1)
+		}
```

Comment:
this can probably be a utility function

Reply:
I'm not sure what "this" is in the current context. I don't see any particular reason to break all this out.

Reply:
I meant the image pull logic, just to save some lines. but more of a nit of course

Reply:
Sorry I still don't understand. If there is something you'd like to change, given that I'm not understanding, could you post a patch or make a change directly to my branch?